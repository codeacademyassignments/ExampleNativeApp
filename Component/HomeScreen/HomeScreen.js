import React from 'react';
import {
  Button, View, StyleSheet, Text,
} from 'react-native';
import PropTypes from 'prop-types';

const style = StyleSheet.create({
  outterMostView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  viewForTopButton: {
    width: 150,
    height: 50,
    backgroundColor: 'powderblue',
  },
  viewForMiddleButton: {
    width: 100,
    height: 50,
    backgroundColor: 'skyblue',
  },
  viewForBottomButton: {
    width: 150,
    height: 50,
    backgroundColor: 'steelblue',
  },
});


class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Welcome',
  };

  state = {
    reload: '',
  }

  onReload = () => () => {
    this.setState({ reload: 'loading..' });
    setTimeout(() => this.setState({ reload: 'Reloaded' }), 1000);
    setTimeout(() => this.setState({ reload: '' }), 2000);
  }

  render() {
    const { navigate } = this.props.navigation;
    const doNavigate = pageToNavigate => () => navigate(pageToNavigate);
    return (
      <View style={style.outterMostView}>
        <View style={style.viewForTopButton}>
          <Button onPress={doNavigate('Details')} title="Greetings" color="#841584" accessibilityLabel="Learn more about this purple button" />
        </View>
        <View style={style.viewForMiddleButton}>
          <Button onPress={doNavigate('PrintPizzas')} title="Explore" color="#841584" accessibilityLabel="Learn more about this purple button" />
        </View>
        <View style={style.viewForBottomButton}>
          <Button onPress={this.onReload()} title="Reload" color="#841584" accessibilityLabel="Learn more about this purple button" />
        </View>
        <Text>{this.state.reload}</Text>
      </View>
    );
  }
}

HomeScreen.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default HomeScreen;
