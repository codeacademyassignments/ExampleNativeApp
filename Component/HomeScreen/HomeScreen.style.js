import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  outterMostView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  viewForTopButton: {
    width: 150,
    height: 50,
    backgroundColor: 'powderblue',
  },
  viewForMiddleButton: {
    width: 100,
    height: 50,
    backgroundColor: 'skyblue',
  },
  viewForBottomButton: {
    width: 150,
    height: 50,
    backgroundColor: 'steelblue',
  },
});
