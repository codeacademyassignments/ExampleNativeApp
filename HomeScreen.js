class HomeScreen extends Component{
render(){
const {navigate} = this.props.navigation;
return(
<Button
title="Go to Jane's profile"
onPress={() => navigate('Profile', {name: 'Jane'})}
/>
)
}
}
