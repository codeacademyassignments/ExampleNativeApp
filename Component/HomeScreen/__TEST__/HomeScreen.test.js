import renderer from 'react-test-renderer';
import HomeScreen from '../HomeScreen';
import React from 'react';
import {shallow} from 'enzyme';
import '../../../setupTests';


describe('HomeScreen',()=>{
  it('should match old snapshot',()=>{
    const tree = renderer.create(<HomeScreen navigation={{navigate: jest.fn()}}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
describe('HomeScreen: reload',()=>{
  const navigateMock = jest.fn();
  const render = shallow(<HomeScreen navigation= {{navigate: navigateMock}}/>);
  it('should reset the state to starting state',()=>{
    expect(render.instance().state.reload).toEqual('');
    render.instance().onReload();
    expect(render.instance().state.reload).toEqual(''); 
  })
})
describe('HomeScreen: Buttons',()=>{
  const navigateMock = jest.fn();
  const render = shallow(<HomeScreen navigation= {{navigate: navigateMock}}/>);
  it('should call navigateMock with (Details,{})',()=>{
    render.find('[title="Greetings"]').simulate('press');
    expect(navigateMock).toHaveBeenCalledWith('Details');
  })
  it('should call navigateMock with (PrintPizzas,{})',()=>{
    render.find('[title="Explore"]').simulate('press');
    expect(navigateMock).toHaveBeenCalledWith('PrintPizzas');
  })
})