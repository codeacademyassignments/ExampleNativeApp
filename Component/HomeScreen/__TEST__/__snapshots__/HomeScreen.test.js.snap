// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should match old snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "powderblue",
        "height": 50,
        "width": 150,
      }
    }
  >
    <View
      accessibilityLabel="Learn more about this purple button"
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Greetings
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "skyblue",
        "height": 50,
        "width": 100,
      }
    }
  >
    <View
      accessibilityLabel="Learn more about this purple button"
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Explore
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "steelblue",
        "height": 50,
        "width": 150,
      }
    }
  >
    <View
      accessibilityLabel="Learn more about this purple button"
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }
        >
          Reload
        </Text>
      </View>
    </View>
  </View>
  <Text>
    
  </Text>
</View>
`;
